### PR TITLE
[crosslink tool] Implement crosslink tool

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** This PR adds the crosslink tool to the repository. Crosslink will automatically insert any missing replace statements for intra-repository go modules. For more info please refer to the crosslink readme [here](https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/crosslink). 

Currently crosslink is run using the `make crosslink` target. It is setup to only perform non-destrructive actions as replace statements should be additive after initial crosslink on-boarding. If other operations are required such as overwriting or pruning crosslink can be run with those flags separately. 

After running `make crosslink` a missing replace statement in `windowperfcountersreceiver` was inserted.

I have also ran crosslink with the `--overwrite` flag and commited the changes to the go.mod files in `awscloudwatchlogsexporter`, `awsemfexporter` and `awsprometheusremotewriteexporter` as an example of what changes crosslink would make. There are many more `go.mod` files in this repository that could be updated. I did not want to create an initial PR with all of those changes at once though. If this is accepted and merged I can update the rest of the `go.mod` files and create a follow on PR. 

I did not add the execution of crosslink to any other process of the makefile. I was hoping to have a discussion on whether that should be done now or should we leave it isolated so developers can use it as they wish? Typically crosslink should be ran from the repository root to ensure that all submodules are mapped properly. 

**Link to tracking Issue:** #7055

**Testing:** Testing for crosslink can be found in the `opentelemetry-go-build-tools` repository. 

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -307,3 +308,8 @@ multimod-verify: install-tools
 .PHONY: multimod-prerelease
 multimod-prerelease: install-tools
 	multimod prerelease -v ./versions.yaml -m contrib-base
+
+.PHONY: crosslink
+crosslink: install-tools
+	@echo "Executing crosslink"
+	crosslink --root=$(shell pwd)
```
Comment (Aneurysm9) [inline Makefile:315]: `\n` at EOF.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,11 +6,13 @@
 ### 💡 Enhancements 💡
 
 - `k8seventsreceiver`: Add Api_version and resource_version (#8539)
+- Add `make crosslink` target to ensure replace statements are included in `go.mod` for all transitive dependencies within repository (#8822)
 
 ### 🛑 Breaking changes 🛑
 
 - `windowsperfcountersreceiver`: Added metrics configuration (#8376)
 
+
```
Comment (Aneurysm9) [inline CHANGELOG.md:15]:
```suggestion
```

**Review by Aneurysm9 — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
Is there any value in adding some docs around using this new capability, maybe in CONTRIBUTING.md?  The goal being making it easier for contributors (especially new contributors) to about `make crosslink`.  Or is this command something that would only be run in our actions?

**Comment by bryan-aguilar** [comment]
> Is there any value in adding some docs around using this new capability, maybe in CONTRIBUTING.md? The goal being making it easier for contributors (especially new contributors) to about `make crosslink`. Or is this command something that would only be run in our actions?

This is a good callout and something I have been thinking about myself. I think there is definitely value in adding it somewhere. I believe `CONTRIBUTING.MD` has instructions to insert replace statements. Let me review that and add some context to those docs.

**Review by codeboten — APPROVED** [review]

**Comment by jpkrohling** [comment]
PR rebased for 0.48.0.

**Comment by bogdandrutu** [comment]
@bryan-aguilar please fix CI/CD

**Comment by bryan-aguilar** [comment]
@bogdandrutu fixed, sorry about the delay.

**Review by codeboten — APPROVED** [review]

**Pull request merged**